### ISSUE TITLE
Record Kubernetes pod name as flow run metadata to enable easier debugging

### ISSUE THREAD

**Issue by tekumara** [PrefectHQ/prefect#5385]
## Current behavior

When debugging issues, it's not easy to link Kubernetes pods to flow runs in the Prefect UI.

It is currently captured in the first log line, but making it available as metadata of the flow run, located with the flow run details in the UI, and accessible via the Prefect API would be nice. 
 
## Proposed behavior

When a pod is launched by the Kubernetes agent, the pod name is captured and reported to the Prefect API and displayed in the UI.

## Example

This would make it much easier to diagnose issues. It would enable the pod for a given flow run to be identified and inspected using our existing Kubernetes tooling.

**Comment by madkinsz** [comment]
Hey @tekumara, this is pretty feasible as a log but we don't have a great place to store it for display in the UI. Right now, I think we attach the flow run id as a pod label, does that not work for your tooling?

**Comment by tekumara** [comment]
Hi @madkinsz thanks for considering this. We don't capture the pod labels in our tooling. All we capture are the kubernetes events, eg:

```
$ kubectl get events
LAST SEEN   TYPE      REASON             OBJECT                           MESSAGE
12m         Normal    Killing            pod/dask-root-a991ffe4-985fhm    Stopping container dask-worker
12m         Normal    Completed          job/prefect-job-ec234a74         Job completed
```

So being able to link via the pod name (eg: `prefect-job-ec234a74`) back to the flow run in Prefect would be super useful when something has gone wrong.

**Comment by madkinsz** [comment]
Hey @tekumara, have you tried using our artifacts API? https://docs.prefect.io/orchestration/concepts/artifacts.html

You can send the pod name (and anything else you'd like) for display in the UI.

**Comment by tekumara** [comment]
Ah yep that would work for the UI. Might be a little hard to parse the markdown contents though via the API. A json artifact would be nice!

**Comment by madkinsz** [comment]
cc @znicholasbrown interesting use-case for a JSON rendered artifact instead of markdown? Although we can just encourage code fences with JSON highlighting within markdown 🤷‍♀️ 

We should do something like this in Prefect 2, cc @WillRaphaelson I think this fits into the observability story. I'm going to close this in favor of a new issue as this will not be changed in Prefect 1.

**Comment by znicholasbrown** [comment]
I think this might work nicely with a 2-column table artifact as well

**Comment by WillRaphaelson** [comment]
Yeah if you open it @madkinsz feel free to assign me, I think this is a good story to tackle either by adding execution environment information to flow run state change events in cloud, or more holistically when we instrument agents/workers to emit telemetry
